src/hooks/wallets/wallets.ts

```diff
@@ -97,3 +99,24 @@ export const isSmartContractWallet = async (wallet: ConnectedWallet) => {
 export const shouldUseEthSignMethod = (wallet: ConnectedWallet): boolean => {
   return isHardwareWallet(wallet) || isSafeMobileWallet(wallet)
 }
+
+type SigningMethods = Parameters<Safe['signTransaction']>[1]
+
+export const getSupportedSigningMethods = (
+  safeVersion: SafeInfo['version'],
+  wallet: ConnectedWallet,
+): SigningMethods[] => {
+  const EIP712_METHODS: SigningMethods[] = ['eth_signTypedData_v4', 'eth_signTypedData_v3', 'eth_signTypedData']
+
+  const safeSupportsEthSign = !!safeVersion && semverSatisfies(safeVersion, '>=1.1.0')
+  if (!safeSupportsEthSign) {
+    return EIP712_METHODS
+  }
+
+  if (shouldUseEthSignMethod(wallet)) {
```

Comment:
If we do a try-catch anyway, why do we need this hardcoded check for HW wallets and mobile pairing?

Reply:
The MetaMask signing window will popup twice otherwise.

Reply:
How's MM related to HW and mobile pairing?

Reply:
We briefly synced, to recap:
* Let's remove the check and keep only the try-catch to reduce the complexity here
* We remove v3 and v4 and keep only the un-postfixed `eth_signTypedData` (because that's the only true one that the EIP defines)